src/vmm/src/rpc_interface.rs

```diff
@@ -485,6 +514,84 @@ impl RuntimeApiController {
         Ok(())
     }
 
+    /// Updates configuration for the balloon device as described in `balloon_update`.
+    fn latest_balloon_stats(&self) -> result::Result<VmmData, VmmActionError> {
```

Comment:
Please move this to a `Vmm` function. The RPC interface is supposed to only do RPC and only care about commands ordering requirements, it shouldn't have virtualization logic.